lib/lightning_web/live/workflow_live/edit.ex

```diff
@@ -403,12 +403,17 @@ defmodule LightningWeb.WorkflowLive.Edit do
        selection_mode: nil,
        selection_params: %{"s" => nil, "m" => nil},
        workflow: nil,
+       workflow_name: nil,
        workflow_params: %{}
      )}
   end
 
   @impl true
   def handle_params(params, _url, socket) do
+    socket =
+      socket
+      |> assign(workflow_name: params["name"])
```

Comment:
since `params` is always read (available on query params) and passed to `apply_action` via `workflow_name`, apply_action can read it directly from params.